flatland_server/include/flatland_server/message_server.h

```diff
@@ -9,210 +9,111 @@
 
 namespace flatland_server {
 
+template <class T>
+class Subscriber;
+template <class T>
+class Publisher;
 class MessageServer;
-class MessageTopicBase {
- public:
-  virtual void clean_old_messages() = 0;
-};
-class PublisherBase {};
-class SubscriberBase {};
+class MessageTopicBase {};
 
 template <class T>
-class MessageView {
-  using const_iterator =
-      typename std::deque<std::pair<ros::Time, T>>::const_iterator;
-
-  const_iterator begin_;
-  const_iterator end_;
-
-  void filterTimeRange(const ros::Time begin, const ros::Time end);
+class MessageTopic : public MessageTopicBase {
+  std::vector<Subscriber<T>> subscribers_;
 
- public:
-  const_iterator begin() const { return begin_; }
-  const_iterator end() const { return end_; }
-
-  bool empty() const { return begin_ == end_; }
-  unsigned size() const;
-
-  MessageView() {}
-  MessageView(const std::deque<std::pair<ros::Time, T>>& t)
-      : begin_(t.begin()), end_(t.end()) {}
-  MessageView(const std::deque<std::pair<ros::Time, T>>& t,
```

Comment:
You can probably delete the #include <deque>

Reply:
yep cool